src/sentry/models/group.py

```diff
@@ -478,17 +478,34 @@ def get_oldest_event_for_environments(self, environments=()):
             project_id=self.project_id,
         )
 
+    def __get_release(self, project_id, group_id, first=True):
+        from sentry.models import GroupRelease, Release
+
+        orderby = "first_seen" if first else "-last_seen"
+        cache_key = f"g-r:{group_id}-{project_id}-{first}"
+        try:
+            release_version = cache.get(cache_key)
+            if not release_version:
+                release_version = Release.objects.get(
+                    id=GroupRelease.objects.filter(group_id=group_id, project_id=project_id)
+                    .order_by(orderby)
+                    .values("release_id")[:1]
+                ).version
+                cache.set(cache_key, release_version, 3600)
+            return release_version
+        except Release.DoesNotExist:
+            cache.set(cache_key, None, 3600)
```

Comment:
Set this to `False`, then you can do `if release_version is not None` above. That way we cache the negative lookup.